### PR TITLE
Documentation on how to avoid pushing directly to RobotLocomotion/drake

### PR DESCRIPTION
**Pull request by amcastro-tri**

### CODE REVIEW

**Comment by amcastro-tri** [comment]
I tried to include @liangfok's and @sherm1's suggestions on the naming convention for the repositories as well. Hopefully it is well explained.

**Comment by RussTedrake** [comment]
thanks!

**Comment by RussTedrake** [comment]
i merged this quickly because I wanted to make sure the pipeline to update the documentation worked (it did).  but reading it more carefully now, I actually don't quite agree with it.  will put my comments inline.

File: drake/doc/no_push_to_origin.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+*****************************************
+Avoid Accidental Pushes to the Repository
+*****************************************
+
+The Issue
+=========
+
+We want to prevent developers from accidentally pushing changes into RobotLocomotion/drake. This is accomplished by:
+
+   1. Everybody agreeing in a naming convention for the respositories.
```
Comment (RussTedrake) [inline drake/doc/no_push_to_origin.rst]: i don't think we need this.  everyone's workflow is different (mine is pretty complicated).

File: drake/doc/no_push_to_origin.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   2. Providing a dummy url for the repository we want to protect.
```
Comment (RussTedrake) [inline drake/doc/no_push_to_origin.rst]: this is not really the central idea.  the core idea (which i didn't know about) is that git can set the "push" target of a remote to be different than the pull target.

File: drake/doc/no_push_to_origin.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+A Convention
+============
```
Comment (RussTedrake) [inline drake/doc/no_push_to_origin.rst]: i would zap this section.

File: drake/doc/no_push_to_origin.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+   "upstream"  = RobotLocomotion/drake
+   "origin" = my fork
+
+so that a simple ``git push`` will push changes into a developer's fork. This still does not prevent you from pushing into ``upstream``. To protect developers from making this simple mistake the following solution is proposed.
+
+A Solution
+==========
+A way to avoid pushing into a repository is providing a dummy url address to the repository. This is accomplished by issuing the git command::
+
+     git remote set-url --push upstream no_push
```
Comment (RussTedrake) [inline drake/doc/no_push_to_origin.rst]: if we do zap the above, then obviously have to change upstream here to origin, etc

**Comment by sherm1** [comment]
The origin/upstream terminology is so common I think it would be reasonable for us to encourage it as a default. For example, [GitHub's help on remotes for forking](https://help.github.com/articles/configuring-a-remote-for-a-fork). The text could acknowledge that other methods can be used, but for developers without a preference we recommend doing it this way.

**Comment by RussTedrake** [comment]
ok.  that's pretty convincing.  :)

**Comment by amcastro-tri** [comment]
I was just starting to work on modifying this so that it would not include
the "upstream" naming and such.
Should I proceed with that or leave the changes accepted as they are right
now?
I guess I could add the github link  in the documentation for those
interested.
I don't use the "upstream" terminology myself but I thought I should
include Sherm's and Liang's suggestion.

Let me know

A.

**Comment by RussTedrake** [comment]
Given this thread, I’m happy to leave it alone unless others weigh in with
different opinions.

**Pull request merged**